canopy_main.ml

```diff
@@ -40,22 +40,22 @@ module Main  (C: CONSOLE) (S: STACKV4) (RES: Resolver_lwt.S) (CON: Conduit_mirag
     in
     let module Store = Canopy_store.Store(C)(Context)(Inflator) in
     let open Canopy_utils in
-    let cache = KeyHashtbl.create 32 in
+    let cache = ref (KeyMap.empty) in
     let open Canopy_config in
     let config = config () in
     let update_atom, atom =
-      Canopy_syndic.atom config Store.last_commit_date cache
+      Canopy_syndic.atom config Store.last_commit_date !cache
```

Comment:
ignore my comment, it works out well! (the `ref Map.empty` returns an address, and while the value is updated, the address stays the same -- thus you can pass the address here, and access will get the filled Map)